### ISSUE TITLE
CMS - 508-defect-3 [COGNITION, SEMANTIC MARKUP]: Download links SHOULD use the pattern documented

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#13819]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

PDF download links **should** use the [pattern used on VA.gov](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340), ([the PR with more info](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/369)). **This is likely to need a new CMS component.** This file currently downloads as `vha-10-10d-fill.pdf`. This is much closer than some other forms. It should match the file indicated in the link. In this case, it may be the file that needs to be renamed to match what the user expects. <br/>_Why it matters:_ The `download` attribute will provide a user-friendly experience by:
    - providing a file name users will be able to identify as the downloaded file;
    - the icon will provide a visual cue that this link behavior is different from other links; 
    - the type attribute tells the browser how to handle this link.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

1. As a user, I want to distinguish download links from regular links so that I may anticipate the behavior.
1. As a user, I want to be able to identify the file I downloaded.
1. As the browser, I depend on the type attribute to know how to handle documents

## Environment

* Operating System: all
* Browser: all
* Server destination: production (but not publicly accessible)

## Steps to Recreate

1. Enter `https://www.va.gov/find-forms/about-form-10-10d/` in browser (use credentials to access)
2. Select the link, "Download VA Form 10-10d (PDF)"
3. Verify the downloaded file name and the link are close but not quite the same
4. Verify the download icon isn't paired with the link
5. Verify the type attribute and the download attribute value is empty, by inspecting the code

## Possible Fixes (optional)

1. Add the type attribute to the download link
1. Add the file name value to the download attribute to name the downloaded file

### Current code

```
<a href="https://www.va.gov/vaforms/medical/pdf/vha-10-10d-fill.pdf" 
    target="_blank" 
    download="" 
    data-widget-type="find-va-forms-invalid-pdf-alert" 
    data-form-number="10-10d" rel="noopener ">
      Download VA Form 10-10d (PDF)
</a>
```

### Recommendation

```diff
<a href="https://www.va.gov/vaforms/medical/pdf/vha-10-10d-fill.pdf" 
+    type="application/pdf" 
    target="_blank" 
+    download="VA Form 10-10d.pdf" 
    data-widget-type="find-va-forms-invalid-pdf-alert" 
    data-form-number="10-10d" rel="noopener ">
+      <i** aria-hidden="true" class="fas fa-download vads-u-padding-right--1" role="img"></i>
      Download VA Form 10-10d (PDF)
</a>
```

## Screenshots or Trace Logs

![Screen Shot 2020-09-21 at 6 27 58 PM](https://user-images.githubusercontent.com/57469/93829651-1b087d80-fc3c-11ea-9299-fc056cd8c37f.png)

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by MarciMcGuireGCIO** [comment]
Assigned to new Decision Tools team; un-assigned from Public Websites

**Comment by MarciMcGuireGCIO** [comment]
Adding blocked tag since this is likely to need a new CMS component and will need to be coordinated with the CMS team.

**Comment by jenniferlee-dsva** [comment]
@MarciMcGuireGCIO - the download links on these pages are not a CMS component. It is FE. The only thing that comes from the CMS is the actual form URL which enters the CMS via the nightly migration. 

**For history:** 
We deliberately added downloading action to the FE  download links. We did this because there are some VA forms that users cannot open in the browser - it's a global beyond VA thing having to do with dynamic XFA PDFs  not being supported by the ISO. So in order to get around that bad experience (users receive a misleading error message -- again, not controled by VA but by Adobe), we automatically initiate a download on clicking the download link. 

**There is one weird thing @ncksllvn and I ran into with ^ however, and that is:** 
VBA forms (forms that start with the number 2x- like 21-526ez, 22-1990, 20-xxxx, etc.) are currently hosted on subdomains. And again, because of browser security -- not a VA thing - our va.gov FE logic doesn't work on subdomains, which basically are treated by browsers like external websites. (At least this is how I was informed.)  

Would love for this to not be the case, if you all find a workaround.

**Comment by MarciMcGuireGCIO** [comment]
Thanks, @jenstrickland.  @ncksllvn, unless you already know the answer - I'll reach out to VSP to ask if cross-origin resource sharing on the server side might be able to remedy the subdomain issue.

**Comment by jenstrickland** [comment]
I wonder if @camerontesterman might have suggestions for those PDFs, and if something can be adjusted in the head. He had to make some adjustments for BAM2 work because of generated PDFs, iirc. The `download` attribute pairs with those settings and, I think, the `type` attribute. Cameron, do you have any ideas for these types of PDFs? 🙇

**Comment by MarciMcGuireGCIO** [comment]
Thanks for the additional info, @jenstrickland. I also posted a question on vsp-platform-support channel related to cross-domain/cross-origin resource sharing.

**Comment by jenstrickland** [comment]
Cameron is the most knowledgable about the PDF header cases, afaik. We sought help before, and he was the one who sorted things out. He's super collaborative!

**Comment by jenniferlee-dsva** [comment]
Thanks all for your due diligence. One more thing I forgot to mention: 

The file name comes directly from the source VA forms database. This doesn't originate in the CMS - it's just one of the many form meta data that's migrated from the VA forms database into the CMS. 

VA has 3 administrations and numerous businesses under each of those administrations - they all have slightly different file naming conventions, and many times, individual form managers don't always follow those conventions. Tl;dr - we don't have any way of controling file names. (This is one of those governance things that we could influence if forms themselves, and not just the meta data, are maintained within the CMS.)

**Comment by jenstrickland** [comment]
The file names can be managed via the header paired with the download attribute. It's possible to use the download link text to populate the download attribute value to assign the actual file name.

**Comment by MarciMcGuireGCIO** [comment]
This will be resolved with the implementation of https://github.com/department-of-veterans-affairs/va.gov-team/issues/22452

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO Can we go about closing this soon?

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel - I plan to close this as soon as Accessibility reviews during the Staging review and confirms its no longer a problem in the updated UI.

**Comment by MarciMcGuireGCIO** [comment]
This was closed as part of applying the new UI pattern.
